### DISCUSSION TITLE
custom css in Jupiter-book

### DISCUSSION THREAD

**Discussion by chipbrock** [executablebooks/meta#387] [Q&A]
Moving from Bookdown to Jupyter books and want to preserve my css stylings from a current project "over there." Otherwise, this is my first experience with the "new" Jupyter books, which looks like a much more robust development approach than Bookdown. Hence, my move.

My understanding is that a custom css file will be respected as the last one loaded if it hangs from the main directory in the _static directory. I've done that with a simple css trial that should only change the font. However, it's not respected in the rendering after "running" it. 

In Bookdown, one would point the apparatus to the custom css with an entry into one of the yml files. Is that not necessary here? I gather that were this not inside of the Jupyter books project, but Sphyx, that one would modify the config.yml file.  

Thanks in advance for suggestions and apologies for any naivety.

**Comment by choldgraf** [comment]
hey! 👋 thanks for the question

could you try following the instructions here: https://jupyterbook.org/advanced/sphinx.html?highlight=custom%20css#custom-css-or-javascript ?

**Reply by chipbrock** [reply]
Thanks for your quick reply. That's actually where I learned to do what I did do. 

But something else has broken as I was playing. Now all three of my installations of books have stopped working with 

`Running Jupyter-Book v0.10.2
Source Folder: /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook
Config Path: /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_config.yml
Output Path: /Users/chipbrock/Dropbox/ImmediateWork/jupyter_books/chipbook/_build/html
Running Sphinx v3.5.4
WARNING: Unknown key in `_toc.yml`: format
WARNING: Unknown key in `_toc.yml`: root
WARNING: Unknown key in `_toc.yml`: options

```
Extension error (jupyter_book.toc):
Handler <function add_toc_to_sphinx at 0x7f9170b41160> for event 'config-inited' threw an exception (exception: 'file')
Traceback (most recent call last):
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/sphinx/events.py", line 111, in emit
    results.append(listener.handler(self.app, *args))
  File "/Users/chipbrock/opt/anaconda3/lib/python3.8/site-packages/jupyter_book/toc.py", line 198, in add_toc_to_sphinx
    app.config["master_doc"] = _posix_no_suffix(toc["file"])
KeyError: 'file'
`
```

Totally clueless about how this could happen, so I'm going to have to open another "help" discussion before working more on the css. Sheesh

**Reply by choldgraf** [reply]
try updating Jupyter Book to v0.11, maybe that would help?

**Reply by chipbrock** [reply]
hmm. There are lots of instructions/tutorials/helpers out there and I guess things are mixing among versions. Likewise, there are multiple ways that I can find to update Jupyter Book. Reinstalling seems not to fix it. What is the preferred way to do the update you're suggesting? I can see that this is moving fast and the internets are slow to keep up. thanks

**Reply by chipbrock** [reply]
I now know what happened in my _toc problem and have updated books.

And I've figured out what was wrong with the local css issue. I'd somehow called` _static/mycss.css _statics/mycss.css`

Thanks for your help.

**Reply by choldgraf** [reply]
Try 

pip install -U jupyter-book
